.grit/patterns/mux_v8.md

```diff
@@ -0,0 +1,278 @@
+---
+title: Upgrade Mux SDK to v8
+---
+
+Upgrade the Mux SDK to v8
+
+tags: #js, #ts, #npm, #upgrade, #mux, #migration
+
+```grit
+engine marzano(0.1)
+language js
+
+pattern convert_config() {
+    object($properties) => $properties where {
+        $properties <: contains bubble pair($key, $value) where or {
+            $key <: `baseUrl` => `baseURL`,
+            $key <: `platform` => `fetch` where {
+                $name = raw``,
+                $version = raw``,
+                $value <: contains bubble($name, $version) pair($key, $value) where or {
+                    $key <: `name` where {
+                        $value <: maybe string(fragment = $val) where {
+                            $name = $val
+                        }
+                    },
+                    $key <: `version` where {
+                        $value <: maybe string(fragment = $val) where  {
+                            $version = $val
+                        }
+                    }
+                },
+                $value => `(url, opts) => {
+                    let opts = opts ?? { headers: {} };
+
+                    opts.headers['x-source-platform'] = '$name | $version';
+
+                    return fetch(url, opts)
+                }`
+            }
+        },
+    }
+}
+
+pattern change_constructors() {
+    `$destruct = new Mux($params)` where {
+        $destruct <: contains `{ $props }` => `mux`,
+        $props <: maybe some change_destructured_property_call(),
+        $params <: or {
+            [$tokenId, $tokenSecret, $config] where {
+                $config <: convert_config() as $parsed_config where {
+                    $params => `{
+                        tokenId: $tokenId,
+                        tokenSecret: $tokenSecret,
+                        $parsed_config
+                    }`
+                }
+            },
+            [$tokenId, $tokenSecret] => `{
+                tokenId: $tokenId,
+                tokenSecret: $tokenSecret,
+            }`,
+            convert_config() as $config => $config,
+            . => .,
+        },
+    },
+}
+
+pattern as_lower_camel_case($formatted) {
+    r"([A-Z])([a-zA-Z]*)"($first_char, $rest) where {
+        $first_char = lowercase(string = $first_char),
+        $formatted = join(list = [$first_char, $rest], separator = ""),
+    }
+}
+
+pattern change_destructured_property_call() {
+    $prop where {
+        $program <: contains bubble($prop) or {
+            `$prop.$field.$action($arg)` => `mux.$prop.$field.$action($arg)`,
+            `$prop.$field.$action($arg, $opt)` => `mux.$prop.$field.$action($arg, $opt)`,
+        } where {
+            $formatted = ``,
+            $prop <: as_lower_camel_case($formatted) where {
```

Comment:
Reusing the same variable is an anti-pattern. I'd just call this `prop_formatted` and use it going forward.